### PR TITLE
Feature ivan hedz elastic search

### PR DESCRIPTION
**Pull request by ihedzde**
Розбив User на User і абстракцію Human. Добавив Elastic й оновив docker-compose додавши туди Elastic Container. Питання таке чи треба комітити Enviroment variable? Якщо так то куди саме (в example)?

### CODE REVIEW

File: backend/src/Infrastructure/InfrastructureDependencyInjection.cs [hunk 1]
```diff
@@ -26,7 +26,20 @@ public static IServiceCollection AddInfrastracture(this IServiceCollection servi
...
+        {
+            var connectionString = Environment.GetEnvironmentVariable("ELASTIC_CONNECTION_STRING");
+            if (connectionString is null)
+                throw new Exception("Elastic connection string url is not specified");
+            var settings = new ConnectionSettings(new Uri(connectionString))
+                .DefaultIndex("defaultIndex")
+                .DefaultMappingFor<User>(m => m
+                .IndexName("users")
+            );
+            services.AddSingleton<IElasticClient>(new ElasticClient(settings));
```
Comment (anton-matsyshyn) [inline backend/src/Infrastructure/InfrastructureDependencyInjection.cs:42]: Singletone may cause multithreading issues, better use Scoped

File: .docker/docker-compose.yml [hunk 2]
```diff
@@ -12,7 +12,20 @@ services:
       - backend
     env_file:
       - ../.env/ats_api.env
-
+  ats_elastic_server:
```
Comment (Yaroslaww-1) [inline .docker/docker-compose.yml:15]: Please also update ```docker-compose.prod.yml``` and ```docker-compose.services.yml```

File: backend/src/Infrastructure/InfrastructureDependencyInjection.cs [hunk 3]
```diff
@@ -26,7 +26,20 @@ public static IServiceCollection AddInfrastracture(this IServiceCollection servi
             services.AddReadRepositories();
 
             services.AddScoped<IDomainEventService, DomainEventService>();
-
+            services.AddElasticEngine();
+            return services;
+        }
+        private static IServiceCollection AddElasticEngine(this IServiceCollection services)
+        {
+            var connectionString = Environment.GetEnvironmentVariable("ELASTIC_CONNECTION_STRING");
```
Comment (Yaroslaww-1) [inline backend/src/Infrastructure/InfrastructureDependencyInjection.cs:34]: Please provide an example of ```ELASTIC_CONNECTION_STRING``` in ```.env.example/ats_api.env```

File: backend/src/Infrastructure/InfrastructureDependencyInjection.cs (hunk 1 above)
Comment (ihedzde) [reply backend/src/Infrastructure/InfrastructureDependencyInjection.cs:42]: It's ok. Singleton in Elastic is thread safe.

**Comment by ihedzde** [comment]
I've made some changes. Think It's ready for approval.

**Comment by ihedzde** [comment]
Fixed merge conflicts: Can be merged in master.

**Comment by ihedzde** [comment]
I've tested search queries in sandbox - working fine. Also added some Entities.

**Review by anton-matsyshyn — APPROVED** [review]

**Pull request merged**